local/rest_api_gcbm/gcbm.py

```diff
@@ -226,25 +247,50 @@ def add_file(self, file_path: str):
         Parameters
         ==========
         1. file_path (str), no default
+        2. category (str), default = "", if skipped - then the categories will be deduced from the file path 
         """
 
         # TODO: update to accept input from Flask endpoint
+        # FIXME: The flask end point should do the pre-processing to be able to only pass the `file_path` as a string
+        # TODO: update in app.py to send valid data to add_file
         filename = os.path.basename(file_path)
         shutil.copy(file_path, os.path.join(self.dirpath, filename))
 
-        if self.disturbances._append(filename):
-            self.disturbances._update_config()
-            return
-        if self.classifiers._append(filename):
-            self.classifiers._update_config()
-            return
-        if self.miscellaneous._append(filename):
-            self.miscellaneous._update_config()
-            return
-        # TODO: Add covariates here
+        def _disturbance(filename, check=True):
+            if self.disturbances._append(filename, check):
+                self.disturbances._update_config()
+                
+        def _classifier(filename, check=True):
+            if self.classifiers._append(filename, check):
+                self.classifiers._update_config()
+
+        def _miscellaneous(filename, check=True):
+            if self.miscellaneous._append(filename, check):
+                self.miscellaneous._update_config()
```

Comment:
@ankitaS11 - this is a really important section. 

Each added file needs to be appended to various sections of the boilerplate `template` config. You can identify the changes that need to be made by searching for the input filenames in our reference simulations.

For example:
```sh
# GCBM Belize
git checkout gcbm-container
grep -rnw GCBM.Belize/Standalone_GCBM/gcbm_project -e "LifeZone"

> ./provider_config.json:15:                    "name": "LifeZone",
> ./variables.json:119:                    "LifeZone"
> ./variables.json:204:        "LifeZone": {
> ./variables.json:209:                "data_id": "LifeZone"
```

For example:
```bash
# FLINT.cloud
git checkout linux-test
cd local/rest_api_gcbm/tests/
unzip linux-demo.zip
grep -rnw config -e "gcbm_input.db"

> config/provider_config.json:4:            "path": "../input_database/gcbm_input.db",
```

Reply:
A bit unsure about this, as in I didn't get what we need to do here but maybe it's just my health that I'm unable to make sense of this. Allow me a day, and I'll come up with specific questions if any, or else I'll fix this in the next commit. :)

Reply:
Hi @ankitaS11  - sorry if the wall of information was overwhelming. 

My goal was to show all the locations where information about the input files are added to `provider_config.json`, `variables.json` and `modules_cbm.json`. You can observe the expected behaviour for each input file type.

- [ ] input files are added to `provider_config.json`
- [ ] input files are given `names` and `layer_prefixes` (except `input_db`)
- [ ] input files are added to `variables.json` and given `data_ids`
- [ ] disturbances are added to `modules_cbm.json`

For now it is fine to ignore the reference to `mean_annual_temperature` in `spinup.json` - but be warned, we will experience errors if no input called `mean_annual_temperature` is uploaded.

Reply:
Thank you, @aornugent - this was helpful! I've attempted to fix this in the commit, could you please take a look?